modules/wkt/src/lib/encode-wkb.ts

```diff
@@ -0,0 +1,321 @@
+// loaders.gl, MIT license
+// Forked from https://github.com/cschwarz/wkx under MIT license, Copyright (c) 2013 Christian Schwarz
+
+import type {
+  Feature, 
+  Geometry, 
+  Point, 
+  MultiPoint, 
+  LineString, MultiLineString,
+  Polygon, 
+  MultiPolygon,
+  GeometryCollection
+} from '@loaders.gl/schema';
+
+import BinaryWriter from './utils/binary-writer';
+
+enum WKB {
+  Point = 1,
+  LineString = 2,
+  Polygon = 3,
+  MultiPoint = 4,
+  MultiLineString = 5,
+  MultiPolygon = 6,
+  GeometryCollection = 7
+};
+
+/**
+ * Encodes a GeoJSON object into WKB
+ * @param geojson A geojson feature or geometry
+ * @returns string
+ */
+export default function encodeWKB(geometry: Geometry | Feature, options?): ArrayBuffer {
+  if (geometry.type === 'Feature') {
+    geometry = geometry.geometry;
+  }
+
+  switch (geometry.type) {
+    case 'Point':
+      return encodePoint(geometry as Point);
+    case 'LineString':
+      return encodeLineString(geometry as LineString);
+    case 'Polygon':
+      return encodePolygon(geometry as Polygon);
+    case 'MultiPoint':
+      return encodeMultiPoint(geometry as MultiPoint);
+    case 'MultiPolygon':
+      return encodeMultiPolygon(geometry as MultiPolygon);
+    case 'MultiLineString':
+      return encodeMultiLineString(geometry as MultiLineString);
+    case 'GeometryCollection':
+      return encodeGeometryCollection(geometry as GeometryCollection);
+    default:
+      throw new Error('stringify requires a valid GeoJSON Feature or geometry object as input');
+  }
+}
+
+/** Calculate the binary size (in the WKB encoding) of a specific geojson geometry */
+function getGeometrySize(geometry: Geometry): number {
+  switch (geometry.type) {
+    case 'Point':
+      return getPointSize(geometry as Point);
```

Comment:
Do you need these casts? I would've expected TS to understand this discriminated union in the `case`